SERVER/src/csm.py

```
@@ -1,40 +1,141 @@
+
+        # 5. Save to Database (DB calls wrapped in to_thread)
+        if narration_text or character_texts:
+            # print("CSM: Saving story turn to database...")
+            await asyncio.to_thread(self.db.save_story, narration_text, character_texts, narrator_audio_path_db)
+            # print("CSM: Story turn processed and saved.")
+        # else:
+            # print("CSM: Nothing to save for this story turn.")
+
+        return narration_text, character_texts
+
+    def update_last_narration_text(self, new_text):
```

Ruff (0.11.9)

Missing return type annotation for public function `update_last_narration_text`

Add return type annotation: `bool`

(ANN201)